src/packet/signature.js

```diff
@@ -674,6 +669,12 @@ class SignaturePacket {
       throw new Error('Public key algorithm used to sign signature does not match issuer key algorithm.');
     }
 
+    this.rawNotations.forEach(({ name, critical }) => {
+      if (critical && (config.knownNotations.indexOf(name) < 0)) {
+        throw new Error(`Unknown critical notation: ${name}`);
+      }
+    });
```

Comment:
Most of the similar checks are at the end of this function. That way, we throw the "most serious" error first. I would do the same here (so that we don't throw this error, which sounds relatively mild, instead of "Signature verification failed", for example).

Reply:
Right, however it seems a waste to do signature verification if we know it's still invalid?

Reply:
I agree, it's inefficient, but it just means that we will be showing an error slower :) I don't think it's a super important case to optimize.